decompiler/frontend/binaryninja/handlers/globals.py

```diff
@@ -76,9 +76,9 @@ def _lift_constant_type(
     ) -> StringSymbol:
         """Lift constant data type (bninja only uses strings) into code"""  # jump table ist auch constant
         if str(variable).find("char const") != -1:
-            string = str(variable.value)[2:-1].rstrip(
-                "\\x00"
-            )  # we want to keep escaped control chars (\n), therefore we take the raw string representation of bytes and purge b""
+            string = str(variable.value.rstrip(b"\x00"))[
+                2:-1
+            ]  # we want to keep escaped control chars (\n), therefore we take the raw string representation of bytes and purge b""
```

Comment:
May I sneek in a little fix here for these lines? Stripping the quotation marks from the binary representation was always quiet lazy. Consider this example:
```python
>>> print(test)
this
is
a
test
>>> print(test.encode("unicode_escape").decode())
this\nis\na\ntest
```

Reply:
I knew about the `unicode_escape` field, but it does not work for decode from bytes. 
Is there something which escapes chars for a string via decode?

Your solution would work, but it'll still be a bit lazy/hacky, because we need to decode encode and decode again. :D 
E.G.: `string = variable.value.rstrip(b"\x00").decode().encode("unicode_escape").decode()`

Reply:
Well I'll find a better solution some day. Still not kinda satisfied with this, but it'll make do.